### PR TITLE
MEN-7616,MEN-7567: subtenant form improvement; Common list and thunk tests

### PR DESCRIPTION
**Pull request by mineralsfree**

### CODE REVIEW

File: frontend/src/js/components/tenants/TenantPage.tsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const TenantPage = () => {
       <h2>Tenants</h2>
       {tenants.length ? <TenantList /> : <TenantsEmptyState openModal={() => setShowCreate(true)} />}
       <Chip color="primary" icon={<AddIcon />} label="Add tenant" onClick={() => setShowCreate(true)} />
-      <TenantCreateForm open={showCreate} onCloseClick={() => setShowCreate(false)} />
+      {showCreate && <TenantCreateForm open={showCreate} onCloseClick={() => setShowCreate(false)} />}
```
Comment (mzedel) [inline frontend/src/js/components/tenants/TenantPage.tsx:53]: doesn't the `open` prop handle the visibility of the form already?

File: frontend/src/js/store/organizationSlice/thunks.ts [hunk 2]
```diff
@@ -221,8 +221,12 @@ interface editTenantBody {
...
   return Api.put(`${tenantadmApiUrlv2}/tenants/${id}/child`, { device_limit: newLimit, name })
     .catch(err => commonErrorHandler(err, `Device Limit cannot be changed`, dispatch))
-    .then(() => Promise.resolve(dispatch(setSnackbar({ message: 'Device Limit was changed successfully', autoHideDuration: TIMEOUTS.fiveSeconds }))))
-    .then(() => dispatch(getTenants()));
+    .then(() => {
+      const tasks = [Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully')))];
+      tasks.push(dispatch(getTenants()));
+      tasks.push(dispatch(getUserOrganization()));
+      return Promise.all(tasks);
+    });
```
Comment (mzedel) [inline frontend/src/js/store/organizationSlice/thunks.ts]:
```suggestion
    .then(() => {
      const tasks = [Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully'))), dispatch(getTenants()), dispatch(getUserOrganization())];
      return Promise.all(tasks);
    });
```
as the `tasks` won't get touched in any other way?
or even:
```suggestion
    .then(() => Promise.all([Promise.resolve(dispatch(setSnackbar('Device Limit was changed successfully'))), dispatch(getTenants()), dispatch(getUserOrganization())]));
```

File: frontend/src/js/components/tenants/TenantCreateForm.tsx [hunk 3]
```diff
@@ -136,23 +146,28 @@ export const TenantCreateForm = (props: TenantCreateFormProps) => {
...
+      await dispatch(addTenant({ name, admin: { password, email }, sso, device_limit: Number(device_limit), binary_delta }));
+    }
     onCloseClick();
   };
   return (
     <Drawer open={open} onClose={onCloseClick} anchor="right" PaperProps={{ style: { minWidth: '67vw' } }}>
       <div className="flexbox center-aligned space-between">
         <div className="flexbox center-aligned">
-          <h3 className={`${classes.tenantTitle}`}>Add a tenant</h3>
+          <h3 className={classes.tenantTitle}>Add a tenant</h3>
```
Comment (mzedel) [inline frontend/src/js/components/tenants/TenantCreateForm.tsx:170]: Thank you!

**Review by mzedel — DISMISSED** [review]

File: frontend/src/js/components/tenants/TenantPage.tsx (hunk 1 above)
Comment (mineralsfree) [reply frontend/src/js/components/tenants/TenantPage.tsx:53]: Yes, but it's an easy way to reset form state :sweat_smile:

**Review by mzedel — APPROVED** [review]

**Pull request merged**